pkg/plugin/test-keeper/plugin/comment_message.go

```diff
@@ -30,6 +31,8 @@ const (
 		"path to an existing file in this repository."
 
 	sadIke = `<img align="left" src="https://cdn.rawgit.com/bartoszmajsak/ike-prow-plugins/2025328b70bd1879520411b3cacadee61a49641a/docs/images/arquillian_ui_failure_128px.png">`
+
+	fileRegex = `(?m)^(?:https?\:)?([a-z_\-\s0-9\.\/]+)+\.(txt|md|doc|docx|adoc)$`
```

Comment:
These messages are only embedded as comments on the PR, and so far it's only Markdown which is supported by GitHub, so I would say let's have a regex for that, but then I would suggest having a unit tests just for this functionality to see if we can cover different cases (http, https, relative path etc etc).

> How about checking if the string (inside of the config file) contains at least one empty space? I think that should be sufficient for detecting if it is an inline string.

As much as it sound likely to cover 99% of the cases, you can always have ppl liking spaces in the file names. Try `wget 'https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/blob/bartoszmajsak-license-fix/interesting space.md'` :)

Reply:
> These messages are only embedded as comments on the PR, and so far it's only Markdown which is supported by GitHub, so I would say let's have a regex for that

but you can reference a file that doesn't have any extension but contains Markdown content

> As much as it sound likely to cover 99% of the cases, you can always have ppl liking spaces in the file names. Try wget 'https://github.com/bartoszmajsak/wfswarm-booster-pipeline-test/blob/bartoszmajsak-license-fix/interesting space.md' :)

Haven't thought about such a ... URL. Ok, so the other option I had in my mind was to validate the URL - either absolute or relative - and those ones that are not valid URLs consider to be the inline messages.

Reply:
> Haven't thought about such a ... URL. Ok, so the other option I had in my mind was to validate the URL - either absolute or relative - and those ones that are not valid URLs consider to be the inline messages. 

I thought of this as well but consider the case `http/server.com/custom_message_file.md`.  This is an invalid URL but it isn't an inline comment.

Reply:
sure, the `http/server.com/custom_message_file.md` is a relative path to a project file. but then the valid URL is constructed by `ghFileService.GetRawFileURL(configuration.PluginHint)` which can be easily validated...

Reply:
> but you can reference a file that doesn't have any extension but contains Markdown content

Let's try to be pragmatic here (also for our own sake). Can we agree on the convention that we only allow `.md` files? And document it too. Maybe that would be good enough to start?

Reply:
> Let's try to be pragmatic here (also for our own sake). 

The same I could say about the spaces in the links ;-)

If you want to be strict and allow only files with the extensions, then I'm fine with it if it is properly documented.
as for the extension - `.md` or `.markdown` should be sufficient. 
For more information about all possible markdown extensions see this  https://superuser.com/questions/249436/file-extension-for-markdown-files (didn't know all these options :-))

Reply:
I have a feeling that we might be drifting in the dangerous direction of making things too configurable. This can backfire some point :)

Maybe we can make it "convention over configuration", so the other way around - like with [GitHub templates](https://help.github.com/articles/creating-an-issue-template-for-your-repository/).

Let's say we expect to have `PLUGIN_NAME_HINT.md` (.txt etc etc file, case insensitive) and nothing else (of course with the current behavior of relative path and URL.

Then as a follow-up story, we can go with the approach of having `.ike-prow/` folder where we expect all our config to be kept. This will bring also some structure and make things easier in my opinion. There are few other aspects such as common settings shared across multiple repos, but I think this we can flesh out in this story too.

Thoughts?

Reply:
I'm fine with it

Reply:
I too like the idea of having something similar to [GitHub templates](https://help.github.com/articles/creating-an-issue-template-for-your-repository/).